docs/spec_draft.md

```diff
@@ -874,6 +875,46 @@ implementation-defined value.
 
 [Back to Ops](#index-of-documented-ops)
 
+## stablehlo.reshape
+
+`stablehlo.reshape(operand) -> result`
+
+### Semantics
+
+Performs reshape of `operand` tensor to a `result` tensor. Conceptually, it
+collapses the dimensions of `operand` starting from slowest-varying
+dimension `0` to fastest-varying dimension `rank-1`, and then refines the
+flatenned vector into the shape of  `result`.
```

Comment:
With the terminology from #193, this explanation could be rewritten as follows:

```
Performs reshape of `operand` tensor to a `result` tensor. Conceptually, it
amounts to keeping the same elements but potentially changing the shape, e.g. 
from `tensor<2x3xf32>` to `tensor<3x2xf32>` or `tensor<6xf32>`.

More formally, the canonical representation of the `result` tensor must be
the same as the canonical representation of the `operand` tensor, which means
that the element types and the number of elements in these tensors must be the same.
```

Reply:
Upon further thinking, I don't like the declarative style of the second paragraph. I think that changing it to imperative style would be better:

```
Performs reshape of `operand` tensor to a `result` tensor. Conceptually, it
amounts to keeping the same canonical representation but potentially changing 
the shape, e.g. from `tensor<2x3xf32>` to `tensor<3x2xf32>` or `tensor<6xf32>`.

More formally, `result[i0, ..., iR-1]` is equal to `operand[j0, ..., jR'-1]`
where `i` and `j` have the same position in the lexicographic ordering of
the index spaces of `result` and `operand`.
```